lib/ProductOpener/Food.pm

```diff
@@ -1297,12 +1380,70 @@ sub compute_nutrition_score ($product_ref) {
 					add_tag($product_ref, "misc", "en:nutriscore-not-applicable");
 					$product_ref->{nutrition_score_debug} = "no nutriscore for category $category_id" . " - ";
 					$product_ref->{nutriscore_data} = {nutriscore_not_applicable_for_category => $category_id};
+					$nutriscore_applicable = 0;
 					last;
 				}
 			}
 		}
 	}
 
+	return $nutriscore_applicable;
+}
+
+=head2 check_availability_of_nutrients_needed_for_nutriscore ($product_ref)
+
+Check that we know or can estimate the nutrients needed to compute the Nutri-Score of the product.
+
+=head3 Return values
+
+=head4 $nutrients_available 0 or 1
+
+=head4 $prepared "" or "_prepared"
+
+Suffix to indicate if the Nutri-Score should be computed on prepared values
+
+=head4 $nutriments_field "nutriments" or "nutriments_estimated"
+
+Indicates which nutrients fields were used to compute the Nutri-Score.
+
+=cut
+
+sub check_availability_of_nutrients_needed_for_nutriscore ($product_ref) {
+
+	my $nutrients_available = 1;
+
+	# do not compute a score for dehydrated products to be rehydrated (e.g. dried soups, powder milk)
```

Comment:
It would be nice to send/use the debug message for knowledge panels

Reply:
the debug message is not meant for users, not localized, not even in English, if we want to convey more data to users, then it needs to be from the structured data